openlabcmd/openlabcmd/zk.py

```diff
@@ -84,19 +84,29 @@ def lost(self):
             return True
         return self.client.state == KazooState.LOST
 
-    def connect(self, hosts=None, read_only=False):
+    def connect(self, hosts=None, timeout=None, read_only=False):
         if not hosts:
             if not self.config:
                 raise exceptions.ClientError('Either config object or hosts '
                                              'string should be provided.')
-            else:
-                try:
-                    hosts = self.config.get('ha', 'zookeeper_hosts')
-                except configparser.NoOptionError:
-                    raise exceptions.ClientError(
-                        "The config doesn't [ha]zookeeper_hosts option.")
+            try:
+                hosts = hosts or self.config.get('ha', 'zookeeper_hosts')
+            except configparser.NoOptionError:
```

Comment:
nit(It's not the problem in this change): Should we also add NoSectionError in here.
[1] https://github.com/python/cpython/blob/45d75fa/Lib/configparser.py#L782